components/playlist/browser/media_detector_component_manager.cc

```diff
@@ -138,11 +152,12 @@ void MediaDetectorComponentManager::OnGetScripts(
     }
   }
 
-  for (auto& observer : observer_list_)
+  for (auto& observer : observer_list_) {
     observer.OnScriptReady(base_script_);
+  }
 }
 
-void MediaDetectorComponentManager::SetUseLocalScriptForTesting() {
+void MediaDetectorComponentManager::SetUseLocalScript() {
```

Comment:
isn't this already handled in `RegisterIfNeeded`? Why do we need to call `OnGetScripts` here as well?